### PR TITLE
[neox-2.x] store current root hash

### PR DESCRIPTION
**Pull request by KickSeason**
### Improve

add `StateRootHashIndex` to store current root

@shargon  Can you please review this?

### CODE REVIEW

File: neo/Persistence/LevelDB/DbSnapshot.cs [hunk 1]
```diff
@@ -47,6 +48,15 @@ public DbSnapshot(DB db)
             ValidatorsCount = new DbMetaDataCache<ValidatorsCountState>(db, options, batch, Prefixes.IX_ValidatorsCount);
             BlockHashIndex = new DbMetaDataCache<HashIndexState>(db, options, batch, Prefixes.IX_CurrentBlock);
             HeaderHashIndex = new DbMetaDataCache<HashIndexState>(db, options, batch, Prefixes.IX_CurrentHeader);
+            StateRootHashIndex = new DbMetaDataCache<HashIndexState>(db, options, batch, Prefixes.IX_CurrentStateRoot,
+            () =>
+            {
+                return new HashIndexState
+                {
+                    Index = 0,
+                    Hash = UInt256.Zero,
```
Comment (shargon) [inline neo/Persistence/LevelDB/DbSnapshot.cs:57]: If we want to be network dependant, we could use the Genesis here, what do you think?

Comment (KickSeason) [reply neo/Persistence/LevelDB/DbSnapshot.cs:57]: I want to store, precisely mark root index here, after verify local root with signed root.
I use this for no local root verified.

Comment (KickSeason) [reply neo/Persistence/LevelDB/DbSnapshot.cs:57]: Do you think we should use genesis hash as the initial or empty state root?

**Comment by KickSeason** [comment]
@shargon You  can merge this, once you approve.

**Comment by KickSeason** [comment]
@erikzhang  Can you merge this one?

**Pull request merged**